src/test/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfigTest.java

```diff
@@ -0,0 +1,414 @@
+                                                                             .setTableName("tableName")
+                                                                             .build(), collector));
+    Schema actualSchema = Schema.recordOf("actualSchema",
+                                          Schema.Field.of("Id", Schema.of(Schema.Type.STRING)));
+
+    Schema providedSchema = Schema.recordOf("providedSchema",
+                                            Schema.Field.of("Id", Schema.of(Schema.Type.STRING)));
+
+    config.getSchema(collector);
+    config.checkCompatibility(actualSchema, providedSchema, collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testCheckCompatibilityWithLogicalType() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = Mockito.spy(withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                             .setTableName("tableName")
+                                                                             .build(), collector));
+    Schema actualSchema = Schema.recordOf("actualSchema",
+                                          Schema.Field.of("CreatedDateTime", Schema.of(Schema.LogicalType.
+                                                                                         TIMESTAMP_MICROS)));
+
+    Schema providedSchema = Schema.recordOf("providedSchema",
+                                            Schema.Field.of("CreatedDateTime", Schema.of(Schema.LogicalType.
+                                                                                           DATETIME)));
+    config.checkCompatibility(actualSchema, providedSchema, collector);
+    Assert.assertEquals(1, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testCheckCompatibility() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = Mockito.spy(withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                             .setTableName("tableName")
+                                                                             .build(), collector));
+    Schema actualSchema = Schema.recordOf("actualSchema",
+                                          Schema.Field.of("Id", Schema.nullableOf(Schema.of(Schema.Type.INT))));
+
+    Schema providedSchema = Schema.recordOf("providedSchema",
+                                            Schema.Field.of("Id", Schema.of(Schema.Type.INT)));
+    config.checkCompatibility(actualSchema, providedSchema, collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+
+  /**
+   * @throws Exception as the apiResponse is unsuccessful, it will not be able to fetch the schema of the table.
```

Comment:
Moved this comment to line 330 where we are catching the ValidationException

Reply:
Done